runtime/src/comm/ugni/comm-ugni.c

```diff
@@ -3564,9 +3564,9 @@ void regMemBroadcast(int mr_i, int mr_cnt, chpl_bool send_mreg_cnt)
       // Update our own map in place.
       //
       if (mr_cnt > 0) {
-        memcpy((char*) &mem_regions_all_my_entry_map[ni]->mregs[mr_i],
-               (char*) &mem_regions->mregs[mr_i],
-               mr_cnt * sizeof(mem_region_t));
+        chpl_memmove((char*) &mem_regions_all_my_entry_map[ni]->mregs[mr_i],
```

Comment:
This cannot overlap; `mem_regions_all_my_entry_map[ni]` points into `mem_regions_all`, and that and `mem_regions` are dynamically allocated, separately.